docs: contributing file fill

romanrostislavovich opened the pull request:
following blocks in `CONTRIBUTING.md` file are fill out:
- `contributing`
- `issue`
- `pull request`

Part of #17

xferra commented:
@svoboda-rabstvo Please close this one or update as soon as #102 was merged.

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
-# CONTRIBUTING
+# Contributing to @linterhub/schema
```
Generic header imho was better: fewer words to change from repo to repo.

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
 
-## Development
+## Contributing
+
```
I'd like to see some `welcome` phrases here if u don't mind.

svoboda-rabstvo replied:
Done

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+If you have any questions, or you've found a bug or want to share any idea,
+please create the relevant issue in this repository.
+
+Please take into account some restrictions:
```
Restrictions, do they?

svoboda-rabstvo replied:
Fix it

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+- Use the GitHub issue search — to ensure that the issue hasn't been already reported.
+- The issue tracker shouldn't be used for personal support requests.
+
+If you want to suggest a Major Feature, please first submit an issue and describe
```
Unclear what is major feature, let's simplify the doc, so potential contributors will not be force to read a lot of rules.

svoboda-rabstvo replied:
Done

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+
 ## Pull Request
 
+Good pull requests - patches, improvements, new features - are a great help.
```
Agai, keep it as small and descriptive as possible, the rest - into PR template.

svoboda-rabstvo replied:
Done

xferra commented on .github/CONTRIBUTING.md:
````diff
@@ -1,28 +1,221 @@
+#### STEP 2: Install
+
+```bash
+git submodule update --init
````
Why submodule step was skipped?

svoboda-rabstvo replied:
Done

xferra commented on .github/CONTRIBUTING.md:
````diff
@@ -1,28 +1,221 @@
+#### STEP 2: Get the latest changes
+
+```bash
+git checkout develop
````
Checkout should be before `npm install`.

svoboda-rabstvo replied:
Done

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+
+Fork the [`linterhub/schema`][github-url] repo and then clone it.
+
+> Note: More information about creating a fork and cloning your fork to local folder read in [`git help`][github-help-fork]
```
How to setup reference to `upstream`? May be some guide?

xferra replied:
@svoboda-rabstvo Not relevant, ty for updates.

xferra commented on .github/CONTRIBUTING.md:
````diff
@@ -1,28 +1,221 @@
+
+```bush
+gulp build
+gulp build-test
````
Incorrect cmd.

svoboda-rabstvo replied:
I will fix it

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+Make sure your commit has the right format:
+
+- The commit message is [properly formatted](#commit-message).
+- The change causes no functional regression. Be sure to run a test to verify your changes before submitting a pull request.
```
Is the checklist a part of the PR template?

xferra replied:
Then let's remove from this doc?

svoboda-rabstvo replied:
Ok

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+
+- A pull request must contain a description explaining what you did and how the effects can be seen.
+
 ### Commit message
```
You can split this doc if it becomes more than 1 page. Just keep it simple.

xferra replied:
and instead of explaining widely used format we can just reference a guide, what do u think?

svoboda-rabstvo replied:
Sounds good, but where to put it?

xferra replied:
If we have a good, already documented, example - just as a link, otherwise - `template` repo (`./docs/`) is the best place IMHO.

xferra replied:
As an example: `template\docs\git\commit_message.md` or smth like that for the explanation below. As an alternative a link to an advanced explanation like https://conventionalcommits.org/ (we can also use exactly this style, why not).

svoboda-rabstvo replied:
done

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+
+[maintainers]: https://github.com/orgs/linterhub/teams/maintainers
+[github-url]: https://github.com/linterhub/schema
+[github-url-issue-bug-reporter]: ../../issues/new?template=Bug_report.md
```
Relative links will not work there, unfortunately.

svoboda-rabstvo replied:
I will fix it

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -1,28 +1,221 @@
+[github-url]: https://github.com/linterhub/schema
+[github-url-issue-bug-reporter]: ../../issues/new?template=Bug_report.md
+[github-url-issue-feature]: ../../issues/new?template=Feature_request.md
+[github-url-issue-custom]: ../../issues/new?template=Custom.md
```
A link to template wizard will be better

svoboda-rabstvo replied:
I will fix it

xferra commented:
@svoboda-rabstvo Do you plan to finish this doc?

svoboda-rabstvo commented:
@xferra Yes of course

svoboda-rabstvo commented on script/gulp/task.lint.js:
```diff
@@ -11,7 +11,15 @@ const config = core.cfg;
+};
 
 // Lint and auto-fix all js files
 const js = () => gulp
```
Yes, for run linting locally

svoboda-rabstvo commented on script/gulp/task.lint.js:
```diff
@@ -11,7 +11,15 @@ const config = core.cfg;
 const markdownlint = core.amd.markdownlint;
 const log = core.amd.log;
 const gulpData = core.amd.gulpData;
-const markdownlintConfig = require(`../../${config.src.config.markdownlint}`);
```
I create a separate issue for it

xferra commented on .github/CONTRIBUTING.md:
```diff
@@ -2,27 +2,216 @@
+
 ### Other questions
 
+You can send you questions in [`@linterhub/maintens`][maintainers] or create
```
I disabled team discussions (GitHub feature), should I re-enable it?

svoboda-rabstvo replied:
Yes, we use that link in another projects

svoboda-rabstvo replied:
I enable discussions

xferra commented on .github/CONTRIBUTING.md:
````diff
@@ -2,27 +2,216 @@
+npm install
+```
+
+#### STEP 5: Create a new branch
````
If steps are too small, maybe numbered list is better than headers `####`?

svoboda-rabstvo replied:
I changed it to `###` and some steps include great description

xferra reviewed (changes requested):
Please review my suggestions above.

romanrostislavovich committed:
docs: fill a contributing file

`CONTRIBUTING.md` the file is filled with following blocks:
- `contributing`
- `issue`
- `pull request`

Part of #17

xferra reviewed (approved):
+1
